src/packet_analysis/Manager.h

```diff
@@ -176,6 +176,13 @@ class Manager : public plugin::ComponentManager<Component> {
      */
     uint64_t GetUnprocessedCount() const { return total_not_processed; }
 
+    /**
+     * Tracks the given analyzer for the current packet's analyzer history.
+     * The packet analyzer history is implemented in form of a stack, which is reset on a
+     * call to ProcessPacket() but maintained throughout calls to ProcessInnerPacket().
+     */
+    void TrackAnalyzer(AnalyzerPtr analyzer) { analyzer_stack.push_back(std::move(analyzer)); }
```

Comment:
Suggest `PushAnalyzer()` as it implies `stack` without needing to check the docs.

Could you add `@analyzer` doxygen documentation?

Reply:
I've added the doxygen documentation.

Regarding renaming the method: I'd argue that the history being a stack could be seen as an implementation detail. I'd prefer calling it `TrackAnalyzer()`. If that's a blocker I'd be okay with `PushAnalyzerToHistory()` as well :)